src/main/java/com/thealgorithms/backtracking/Combination.java

```diff
@@ -1,62 +1,58 @@
 package com.thealgorithms.backtracking;
 
-import java.util.Arrays;
-import java.util.LinkedList;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.TreeSet;
 
 /**
- * Finds all permutations of given array
- * @author Alan Piao (<a href="https://github.com/cpiao3">git-Alan Piao</a>)
+ * Finds all combinations of a given array of unique elements.
  */
 public final class Combination {
     private Combination() {
     }
 
-    private static int length;
-
     /**
-     * Find all combinations of given array using backtracking
-     * @param arr the array.
-     * @param n length of combination
-     * @param <T> the type of elements in the array.
-     * @return a list of all combinations of length n. If n == 0, return null.
+     * Finds all combinations of a given array of length n.
+     * If n is 0, an empty list is returned.
+     *
+     * @param data            the array of elements.
+     * @param combinationSize the desired size of combinations.
+     * @param <T>             the type of elements in the array (assumed to be unique).
+     * @return a list of all combinations of size `combinationSize`.
+     * @throws IllegalArgumentException if combinationSize is negative.
      */
-    public static <T> List<TreeSet<T>> combination(T[] arr, int n) {
-        if (n == 0) {
+    public static <T> List<TreeSet<T>> combination(T[] data, int combinationSize) {
+        if (combinationSize < 0) {
+            throw new IllegalArgumentException("Combination size cannot be negative.");
+        } else if (combinationSize == 0) {
             return null;
         }
```

Comment:
I do not see a reason to return `null` here. You can just remove this _branch_, so the result would be `Arrays.asList(new TreeSet<Integer>())`.

The problem is `ArrayCombination`. The method [`combination` will need an update](https://github.com/TheAlgorithms/Java/blob/26b4b82949ae2afb4e2aab94a2c96ee07acff5aa/src/main/java/com/thealgorithms/backtracking/ArrayCombination.java#L23).